triopg/_triopg.py

```diff
@@ -90,7 +90,8 @@ def cursor(self, *args, **kwargs):
     def __getattr__(self, attr):
         target = getattr(self._asyncpg_statement, attr)
 
-        if callable(target):
+        if iscoroutinefunction(target.__wrapped__
+                               if hasattr(target, '__wrapped__') else target):
```

Comment:
Unfortunately, `inspect.iscoroutinefunction` did not work directly, because `PreparedStatement` methods are wrapped with `@connresource.guarded`.

Reply:
You should add a comment about this then ;-)

Reply:
btw, shouldn't we run the `st.get_query()`&co functions within an asyncio context ?
Even if those function are synchronous, they could call ` asyncio.get_event_loop()` which wouldn't return the correct event loop given trio_asyncio run the asyncio loop in another thread.

Of course this could be also true for any @property, so `target = getattr(self._asyncpg_statement, attr)` should also be run from an asyncio context if we want to be super careful !

So in the end I would say it's ok not to wrap property and sync function in asyncio context call as long as we checked asyncpg doesn't do this kind of trick (and we consider it's unlikely its going to rely on this kind of stuff)